proxy/http/HttpTunnel.cc

```diff
@@ -589,8 +589,8 @@ HttpTunnel::kill_tunnel()
     ink_assert(producer.alive == false);
   }
   active = false;
-  this->deallocate_buffers();
   this->deallocate_redirect_postdata_buffers();
+  this->deallocate_buffers();
```

Comment:
You should check all the calls to `deallocate_redirect_postdata_buffers();` and always call the `deallocate_buffers();` after it.

HttpSM::tunnel_handler_cache_fill, HttpSM::tunnel_handler_100_continue, HttpSM::tunnel_handler_push, etc ...